Implement layer card more menu and several code cleanup tasks.

amcgee opened the pull request:
This is groundwork for [DHIS2-3216](https://jira.dhis2.org/browse/DHIS2-3216).

I implemented a "more" menu in the Layer card toolbar to facilitate adding more options.  Before this change the toolbar was getting crowded, and the addition of `download` and `open in data visualizer` buttons was impossible.

I moved the `Data Table`, `Edit`, and `Remove` layer actions into the expansion menu, we need to decide whether `Edit` should remain a top-level toolbar action (Joe, @turban votes for keeping it top-level).

**Before**

![screenshot 2018-09-20 12 12 51](https://user-images.githubusercontent.com/947888/45812523-4ca68100-bcd0-11e8-82cd-25d3dfde69f0.png)

**After**

![screenshot 2018-09-20 12 25 05](https://user-images.githubusercontent.com/947888/45812498-3bf60b00-bcd0-11e8-81ce-fde1a6eb54b8.png)

In this change I've also performed several code cleanup tasks, including:
* Factor the Layer Toolbar to be used in both LayerCard and BasemapCard to reduce duplication
* Remove *.css files from the `src/components/layers` sub-tree, implementing those styles in JSS instead
* Fix highlighting of the currently selected basemap (id was not passed and so nothing was highlighted)

This PR does not yet implement the download functionality from DHIS2-3216 yet, so it should be merged separately.  Once it is merged, the PR for DHIS2-3216 will be concise and clean.

Another factorization task which I did not undertake would be to de-duplicate the code in `BasemapCard.js` and `LayerCard.js`, since they are very similar except for the Redux connections, but that is a task for another day.

turban commented:
Good work Austin and thanks for improving the code quality :-) We need the expansion menu for new fuctionality as data download  and switching view to charts/pivots. I'm not sure if we should move existing layer actions to this menu as they will be more hidden from the users, although it looks cleaner. Any opinions here Joe?

joeDHIS commented:
I think it makes sense to display the 'Edit' action in the layer toolbar **and** in the expansion menu. 

'Edit' is the most important command here so it's ok that we allow it to be accessed from two areas. We can see that the icon in the toolbar will be used by users who know their way around the app, whereas the 'Edit layer' in the expansion menu is great for helping users who are struggling to find out where to find that functionality.

The 'Data table' and 'Remove' actions should exist only in the expansion menu as these are secondary (and for remove, destructive) actions.

Toolbar actions can be displayed in this order:
Edit, Visibility Toggle (eye), opacity slider......overflow menu (aligned right)

turban commented:
Changes proposed are fine with me!

amcgee commented:
Thanks @joeDHIS, I'll make those changes!

amcgee commented:
PR updated, UI now looks like this (Note that the order of visibility toggle and opacity slider also changes for the Basemap card)

![screenshot 2018-09-20 15 02 16](https://user-images.githubusercontent.com/947888/45820195-9732f800-bce6-11e8-81f3-918ca635db4a.png)

amcgee commented on src/components/edit/EarthEngineDialog.js:
```diff
@@ -70,10 +69,20 @@ const datasets = {
+        margin: 0,
+        marginLeft: -32,
     },
+    // Unused?
```
@turban I left this as as comment for now, feel free to have me remove if if you think it's safe to remove.

turban replied:
The borders are adding extra margin to the "columns" - see screenshot here: https://github.com/dhis2/maps-app/pull/21#issuecomment-423465031

turban replied:
Ref. https://stackoverflow.com/questions/20626685/better-way-to-set-distance-between-flexbox-items
Open for other ways to do the layout positioning :-)

amcgee commented on src/components/layers/basemaps/BasemapList.js:
```diff
@@ -1,25 +1,35 @@
+    container: {
+        maxHeight: 270,
+        overflowY: 'auto',
+        // overflowY: 'overlay', // This is a deprecated API and we can't have both 'overlay' and 'auto', see https://developer.mozilla.org/en-US/docs/Web/CSS/overflow
```
Likewise with this comment.

turban commented:
The colors are not showing in the legend preview (top) - should be like bottom. 
![skjermbilde 2018-09-21 kl 10 57 56](https://user-images.githubusercontent.com/548708/45871267-5510c180-bd8d-11e8-8be7-fdd7f736b6e4.png)
![skjermbilde 2018-09-21 kl 10 57 46](https://user-images.githubusercontent.com/548708/45871268-5510c180-bd8d-11e8-8458-f36d1181b746.png)

turban commented:
Is it possible to add a tooltip to the opacity slider as well?

turban commented:
Missing text on layer thumbs without image. 

![skjermbilde 2018-09-21 kl 11 10 04](https://user-images.githubusercontent.com/548708/45871913-ef253980-bd8e-11e8-8e64-9c9aaf2f0471.png)
![skjermbilde 2018-09-21 kl 11 09 54](https://user-images.githubusercontent.com/548708/45871914-efbdd000-bd8e-11e8-8be0-195cf504786a.png)

Better solution targeted for 2.31: https://jira.dhis2.org/browse/DHIS2-2566

amcgee commented:
PR Updated:

> The colors are not showing in the legend preview

Good catch!  This was because the Earth Engine Dialog was re-using the Legend class from the now-defunct Legend.css.  I refactored it to share style object


> The toolbar should be visisble when scroll is needed - not an issue with this PR - but we could fix it here :-)

I added a container div to the basemapCard so that scrolling works correctly, the issue was with `position: relative` on the basemap card.

> Missing text on layer thumbs without image.

This was an issue with the `lineHeight` property in JSS - it needs to explicitly specify `52px` not `52` since they have different meanings.  Fixed.

> Is it possible to add a tooltip to the opacity slider as well?

Yes, added `Set layer opacity` (with an additional wrapper div to make it appear consistently)

**One question**: I included a tooltip for the expansion menu button which just says `More`.  This isn't very helpful.  Should we remove it or change it to something more verbose and descriptive?  @joeDHIS  (screenshot below)

![screenshot 2018-09-21 14 52 30](https://user-images.githubusercontent.com/947888/45882307-0f182580-bdae-11e8-99cd-8ffb4c6ce6a8.png)

amcgee commented:
🎉 @joeDHIS for the More menu tooltip question above, we'll put that in a small separate change later on if we want to update it.
